maintainers/scripts/update-python-libraries

```diff
@@ -0,0 +1,221 @@
+#! /usr/bin/env nix-shell
+#! nix-shell -i python3 -p 'python3.withPackages(ps: with ps; [ requests toolz unidecode ])'
+
+"""
+Update a Python package expression by passing in the `.nix` file, or the directory containing it.
+You can pass in multiple files or paths.
+
+You'll likely want to use
+``
+  $ ./update-python-libraries ../../pkgs/development/python-modules/*
+``
+to update all libraries in that folder.
+"""
+
+import argparse
+import logging
+import os
+import re
+import requests
+import toolz
+
+INDEX = "https://pypi.io/pypi"
+"""url of PyPI"""
+
+EXTENSIONS = ['tar.gz', 'tar.bz2', 'tar', 'zip', '.whl']
+"""Permitted file extensions. These are evaluated from left to right and the first occurance is returned."""
+
+def _get_value(attribute, text):
+    """Match attribute in text and return it."""
+    regex = '{}\s+=\s+"(.*)";'.format(attribute)
+    regex = re.compile(regex)
+    value = regex.findall(text)
+    if len(value):
+        return value[0]
+    else:
+        return None
+
+def _get_line_and_value(attribute, text):
+    """Match attribute in text. Return the line and the value of the attribute."""
+    regex = '({}\s+=\s+"(.*)";)'.format(attribute)
+    regex = re.compile(regex)
+    value = regex.findall(text)
+
+    return value[0]
+
+def _replace_value(attribute, value, text):
+    """Search and replace value of attribute in text."""
+    old_line, old_value = _get_line_and_value(attribute, text)
+    new_line = old_line.replace(old_value, value)
+    new_text = text.replace(old_line, new_line)
+    return new_text
+
+def _fetch_page(url):
+    r = requests.get(url)
+    if r.status_code == requests.codes.ok:
+        return r.json()
+    else:
+        logging.warning("Request for {} failed".format(url))
+
+def _get_latest_version(package, extension):
```

Comment:
The GitHub API is indeed also very useful.

I've been thinking of putting together a repo where we list for each package where sources can be found. Something like
```
{ "numpy" :  [
  ("GitHub", { repo = ...; owner =... ; tagRegex = "..."; }),
  ("PyPI", { name = "numpy"}),
], "scipy" : [ ...
],
}
```
Then we can use the different fetchers to build up a list of available versions and files.

Or, we don't make such a list, and just pull everything from each and try to put it together, but then you end up with https://libraries.io. I think that's a nice project, although, as with all projects that pull data and interpret it, it makes mistakes.